lib/smack/SmackInstGenerator.cpp

```diff
@@ -623,6 +623,16 @@ void SmackInstGenerator::visitCallInst(llvm::CallInst& ci) {
     Value* val = ci.getArgOperand(0);
     emit(Stmt::assign(rep->expr(&ci), rep->expr(val)));
 
+  } else if (SmackOptions::BitPrecise && SmackOptions::FloatEnabled &&
+    (name.find("llvm.convert.from.fp16.f32") != std::string::npos ||
+    name.find("llvm.convert.from.fp16.f64") != std::string::npos ||
```

Comment:
Could we use `getIntrinsicID()` to do this matching, instead of using strings?

Reply:
It turns out `getIntrinsicID` is tricky to use. Apart from that, it seems this function returns an ID for overloaded intrinsics. For example, all the conversions to `fp16` has one ID. We have to resort to the function names.

Reply:
I see. I have to say I find it hard to review this pull request independently of suggesting more general refactoring changes to how we handle intrinsics. So let's maybe push this one true, and follow it with refactoring right away (before I forget everything I wrote here about refactoring).
Does that sound good?

Reply:
Sounds good to me. I implemented a new version that uses `getIntrinsicID` instead of function names.